diff --git a/internal/conns/config.go b/internal/conns/config.go
index 13b7d153a7..1d981e9097 100644
--- a/internal/conns/config.go
+++ b/internal/conns/config.go
@@ -86,8 +86,23 @@ type Config struct {
 	UseFIPSEndpoint                bool
 }

+// XXX: added by bblommers
+func GetLocalEndpoints() map[string]string {
+	const localEndpoint = "http://localhost:4566"
+	var localEndpoints = map[string]string{}
+	for _, name := range names.Aliases() {
+		localEndpoints[name] = localEndpoint
+	}
+	return localEndpoints
+}
+
 // ConfigureProvider configures the provided provider Meta (instance data).
 func (c *Config) ConfigureProvider(ctx context.Context, client *AWSClient) (*AWSClient, diag.Diagnostics) {
+
+	// XXX: added by bblommers
+	// insert custom endpoints
+	c.Endpoints = GetLocalEndpoints()
+
 	awsbaseConfig := awsbase.Config{
 		AccessKey:                     c.AccessKey,
 		APNInfo:                       StdUserAgentProducts(c.TerraformVersion),
diff --git a/internal/provider/provider.go b/internal/provider/provider.go
index c49bd366d3..2db930e61d 100644
--- a/internal/provider/provider.go
+++ b/internal/provider/provider.go
@@ -2196,13 +2196,13 @@ func configure(ctx context.Context, provider *schema.Provider, d *schema.Resourc
 		CustomCABundle:                 d.Get("custom_ca_bundle").(string),
 		EC2MetadataServiceEndpoint:     d.Get("ec2_metadata_service_endpoint").(string),
 		EC2MetadataServiceEndpointMode: d.Get("ec2_metadata_service_endpoint_mode").(string),
-		Endpoints:                      make(map[string]string),
+		Endpoints:                      conns.GetLocalEndpoints(),  // XXX: added by bblommers
 		HTTPProxy:                      d.Get("http_proxy").(string),
 		Insecure:                       d.Get("insecure").(bool),
 		MaxRetries:                     25, // Set default here, not in schema (muxing with v6 provider).
 		Profile:                        d.Get("profile").(string),
 		Region:                         d.Get("region").(string),
-		S3UsePathStyle:                 d.Get("s3_use_path_style").(bool) || d.Get("s3_force_path_style").(bool),
+		S3UsePathStyle:                 true,
 		SecretKey:                      d.Get("secret_key").(string),
 		SkipCredsValidation:            d.Get("skip_credentials_validation").(bool),
 		SkipGetEC2Platforms:            d.Get("skip_get_ec2_platforms").(bool),
