From 84af701010212ceccccba66422191dfcf1e7445b Mon Sep 17 00:00:00 2001
From: Bert Blommers <info@bertblommers.nl>
Date: Sun, 13 Aug 2023 21:35:11 +0000
Subject: [PATCH] Patch IAM timings

---
 internal/service/iam/wait.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/internal/service/iam/wait.go b/internal/service/iam/wait.go
index 3d1c542aa2..e2e5069b63 100644
--- a/internal/service/iam/wait.go
+++ b/internal/service/iam/wait.go
@@ -20,7 +20,7 @@ const (
 	// as this will negatively impact user experience when configurations
 	// have incorrect references or permissions.
 	// Reference: https://docs.aws.amazon.com/IAM/latest/UserGuide/troubleshoot_general.html#troubleshoot_general_eventual-consistency
-	propagationTimeout = 2 * time.Minute
+	propagationTimeout = 2 * time.Second

 	RoleStatusARNIsUniqueID = "uniqueid"
 	RoleStatusARNIsARN      = "arn"
-- 
2.25.1

