From c2981f42629c1dcb3756c13f243c8c52391f3677 Mon Sep 17 00:00:00 2001
From: Bert Blommers <info@bertblommers.nl>
Date: Sun, 13 Aug 2023 21:43:26 +0000
Subject: [PATCH] Patch: CloudFront timings

---
 internal/service/cloudfront/distribution.go | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/internal/service/cloudfront/distribution.go b/internal/service/cloudfront/distribution.go
index 4870ca0f6d..8190b12231 100644
--- a/internal/service/cloudfront/distribution.go
+++ b/internal/service/cloudfront/distribution.go
@@ -1120,9 +1120,9 @@ func DistributionWaitUntilDeployed(ctx context.Context, id string, meta interfac
 		Pending:    []string{"InProgress"},
 		Target:     []string{"Deployed"},
 		Refresh:    resourceWebDistributionStateRefreshFunc(ctx, id, meta),
-		Timeout:    90 * time.Minute,
-		MinTimeout: 15 * time.Second,
-		Delay:      1 * time.Minute,
+		Timeout:    90 * time.Second,
+		MinTimeout: 2 * time.Second,
+		Delay:      2 * time.Second,
 	}

 	_, err := stateConf.WaitForStateContext(ctx)
-- 
2.25.1

